### PR TITLE
[Snapshot] Add test-without-building capability

### PR DESCRIPTION
**Pull request by igor-makarov**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Allow to build test suite once and run multiple times, to save time on large language sets.

### Description
Add `test_without_building` option that is emitted to the xcodebuild command.

### CODE REVIEW

File: snapshot/lib/snapshot/options.rb [hunk 1]
```diff
@@ -137,6 +137,13 @@ def self.available_options
                                      description: "Should the project be cleaned before building it?",
                                      is_string: false,
                                      default_value: false),
+        FastlaneCore::ConfigItem.new(key: :test_without_building,
+                                     short_option: "-T",
+                                     env_name: "SNAPSHOT_TEST_WITHOUT_BUILDING",
+                                     description: "Test without building, requires a derived data path",
```
Comment (KrauseFx) [inline snapshot/lib/snapshot/options.rb]: Can we add a check for this in `snapshot/detect_values.rb` and show a nice `UI.user_error!("...")` if that's not the case, with instructions on how to set it up?

**Review by KrauseFx — COMMENTED** [review]
This is great, thanks for your contribution @igor-makarov 👍 

One small comment also. And, can we update `snapshot.md` to mention this and how to use it somewhere?

**Comment by igor-makarov** [comment]
I've added the @KrauseFx requested. 

However, I'm having trouble figuring out where in `snapshot.md` to mention this. I think this should go into docs website. Do I need to edit [this page](https://docs.fastlane.tools/actions/capture_ios_screenshots/) somehow, or is it updated automatically?

**Comment by KrauseFx** [comment]
Ah, sorry, I meant this file: https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/docs/snapshot.md which is used to auto-generated the rest of the docs 👍

**Comment by KrauseFx** [comment]
It's then rendered to https://docs.fastlane.tools/actions/snapshot/

File: snapshot/lib/snapshot/detect_values.rb [hunk 2]
```diff
@@ -19,6 +19,10 @@ def self.set_additional_default_values
         end
       end
 
+      if !config[:test_without_building].nil? && config[:derived_data_path].nil?
+        UI.user_error! "Cannot use test_without_building option without a derived_data_path!"
```
Comment (KrauseFx) [inline snapshot/lib/snapshot/detect_values.rb]: Please add `(` around the method call

File: snapshot/lib/snapshot/detect_values.rb [hunk 3]
```diff
@@ -19,6 +19,10 @@ def self.set_additional_default_values
         end
       end
 
+      if !config[:test_without_building].nil? && config[:derived_data_path].nil?
```
Comment (KrauseFx) [inline snapshot/lib/snapshot/detect_values.rb]: Use `config[:derived_data_path].to_s.length == 0` instead, as an empty string should also show this error message

**Comment by igor-makarov** [comment]
I checked the file you mentioned, and there is no param list there. The params are auto-generated from the options.

**Review by KrauseFx — APPROVED** [review]
Thanks for updating 👍

**Pull request merged**